connector/src/yang/connector/netconf.py

```diff
@@ -39,6 +40,57 @@ class BaseConnection:
 # create a logger for this module
 logger = logging.getLogger(__name__)
 
+
+nccl = logging.getLogger("ncclient")
+# The 'Sending' messages are logged at level INFO.
+# The 'Received' messages are logged at level DEBUG.
+
+
+class NetconfSessionLogHandler(logging.Handler):
+    """Logging handler that pretty prints ncclient XML."""
+
+    parser = et.XMLParser(recover=True)
+
+    def emit(self, record):
+        if hasattr(record, 'session'):
+            # If the message contains XML, pretty-print it
+            record.args = list(record.args)
+
+            for i in range(len(record.args)):
+                arg = None
+                if isinstance(record.args[i], str):
+                    arg = record.args[i].encode("utf-8")
+                elif isinstance(record.args[i], bytes):
+                    arg = record.args[i]
+                if not arg:
+                    continue
+                start = arg.find(b"<")
+                end = arg.rfind(b"]]>]]>")    # NETCONF 1.0 message terminator
+                if end == -1:
+                    end = arg.rfind(b">")
+                    if end != -1:
+                        # Include the '>' character in our range
+                        end += 1
+                if start != -1 and end != -1:
+                    try:
+                        elem = et.fromstring(arg[start:end], self.parser)
+                        if elem is None:
+                            continue
+
+                        text = et.tostring(elem, pretty_print=True,
+                                           encoding="utf-8")
+                        record.args[i] = (arg[:start] +
+                                          text +
+                                          arg[end:]).decode()
+                    except:
```

Comment:
```python
except (et.Error, UnicodeError):
```
So that BaseExceptions aren't suppressed. Otherwise looks good.

Reply:
Only thing I can get bad parameters to throw are either TypeError for non-element values sent to et.tostring or AttributeError if .decode() is not there.  Really, TypeError would be the only one to catch that would be relevant to et.tostring.  Why do you suggest et.Error and UnicodeError?

Reply:
Come to think of it, if my new formatting code fails to parse the record for whatever reason, I want _all_ of those errors to be suppressed and allow the record to continue as-is.  I should expand the try: except:.